mlflow/utils/search_utils.py

```diff
@@ -607,16 +655,47 @@ def _parse_filter_for_model_registry(cls, filter_string, valid_search_keys):
                 error_code=INVALID_PARAMETER_VALUE,
             )
         statement = parsed[0]
-        invalids = list(filter(cls._invalid_statement_token, statement.tokens))
+        invalids = list(
+            filter(cls._invalid_statement_token_search_model_registry, statement.tokens)
+        )
         if len(invalids) > 0:
             invalid_clauses = ", ".join("'%s'" % token for token in invalids)
             raise MlflowException(
                 "Invalid clause(s) in filter string: %s. " "%s" % (invalid_clauses, expected),
                 error_code=INVALID_PARAMETER_VALUE,
             )
+        # process non-Comparison tokens
+        if len(statement.tokens) == 1 and isinstance(statement.tokens[0], Comparison):
+            token_list = statement.tokens
+        elif len(statement.tokens) > 1:
+            idx = 0
+            comparison_subtokens = []
+            while idx < len(statement.tokens):
+                if statement.tokens[idx].is_whitespace:
+                    idx += 1
+                    continue
+                elif isinstance(statement.tokens[idx], Comparison):
+                    raise MlflowException(
+                        "Search filter '%s' contains multiple expressions. "
+                        "%s " % (filter_string, expected),
+                        error_code=INVALID_PARAMETER_VALUE,
+                    )
+                if (
+                    isinstance(statement.tokens[idx], Identifier)
+                    or statement.tokens[idx].match(ttype=TokenType.Keyword, values=["IN"])
+                    or isinstance(statement.tokens[idx], Parenthesis)
+                ):
+                    comparison_subtokens.append(statement.tokens[idx])
+                else:
+                    break
+                idx += 1
```

Comment:
would the following be equivalent?
```
            for token in statement.tokens:
                if isinstance(token, Comparison):
                    raise MlflowException(
                        "Search filter '%s' contains multiple expressions. "
                        "%s " % (filter_string, expected),
                        error_code=INVALID_PARAMETER_VALUE,
                    )

                if (
                    isinstance(token, Identifier)
                    or token.match(ttype=TokenType.Keyword, values=["IN"])
                    or isinstance(token, Parenthesis)
                ):
                    comparison_subtokens.append(token)
                elif not token.is_whitespace:
                    break
```

Reply:
It would - thanks so much for the much cleaner refactor